lib/server/controllers/view_controller.rb

```diff
@@ -0,0 +1,128 @@
+class ViewController < Timur::Controller
+
+  # Get the tab view data. The tab view data is a json representation of a
+  # front-end layout.
+
+  def view_json
+    success(pull_view_data.to_json, 'application/json')
+  end
+
+  def update_view_json
+    unless current_user.is_admin?(@params[:project_name])
+      failure(
+        401,
+        {errors: ['You do not have the permissions to edit this data.']}
+      )
+      return
+    end
+
+    if @params.key?(:tabs)
+      @params[:tabs].each do |tab_name, tab_datum|
+        ViewTab.update(
+          @params[:project_name],
+          @params[:model_name],
+          @params[:tabs][tab_name][:name],
+          tab_datum
+        )
+      end
+    else
+      failure(
+        401,
+        {errors: ['There is no tab data to save.']}
+      )
+      return
+    end
+
+    success(pull_view_data.to_json, 'application/json')
+  end
+
+  def delete_view_json
+
+    unless current_user.is_admin?(@params[:project_name])
```

Comment:
You should use the auth: { user: { is_admin?: project_name } } option on routes to do this (it returns a 403, Forbidden).

Reply:
+1